teachertool/src/utils/fileSystemHelpers.ts

```diff
@@ -0,0 +1,31 @@
+import { logError } from "../services/loggingService";
+import { stateAndDispatch } from "../state";
+import { ErrorCode } from "../types/errorCode";
+
+// convert teachertool/somefile.json to teachertool/test/somefile.json
+function getTestFileName(fileName: string) {
+    const parts = fileName.split("/");
+    parts.splice(parts.length - 1, 0, "test");
+    return parts.join("/");
+}
+
+// This allows us to take multiple lists stored in json docs files and concatenate them into
+// a single unified list. It also automatically adds entries from test files if the test catalog flag is set.
+export async function loadTestableCollectionFromFileAsync<T>(fileNames: string[], rootName: string): Promise<T[]> {
```

Comment:
I feel this code didn't end up generalizing very cleanly. I was torn on whether or not to keep the helper function or just have two transforms (loadCatalogAsync and loadValidatorPlansAsync) with very similar code. I ended up keeping it, but I'm happy to switch back if folks feel this is too messy and/or troublesome to maintain. I don't anticipate it being shared outside of those two scenarios.